lib/ffi-libarchive/reader.rb

```diff
@@ -171,5 +185,57 @@ def save_data(file_name)
         raise Error, @archive if C.archive_read_data_into_fd(archive, fd) != C::OK
       end
     end
+
+    private
+
+    #
+    # See:
+    #   1. https://github.com/libarchive/libarchive/blob/6a9dcf9fc429e2dc9fb08e669bf7b0bed4d5edf9/tar/read.c#L346
+    #   2. https://github.com/libarchive/libarchive/blob/a11f15860ae39ecdc8173243a211cdafc8ac893c/tar/util.c#L523-L535
+    #   3. https://github.com/libarchive/libarchive/blob/a11f15860ae39ecdc8173243a211cdafc8ac893c/tar/util.c#L554-L560
+    #
+    # @param entry [Archive::Entry]
+    #
+    # @return [Archive::Entry, nil] entry stripped or nil if entry is no longer relevant due to stripping
+    #
+    def strip_entry_components(entry)
```

Comment:
this should probably just be `strip_entry_components!` since it mutates the entry.

Reply:
@lamont-granquist I spent about 30 minutes trying to decide if this should have a bang or not. In all of the core of Ruby, every time the `!` is used it's `nil` response means it was __not__ modified which this abuses that pattern. If you strongly think it should be a `!` I'll add it back in there.

Reply:
The other option is to change the API and use more composition, adding something like:

```ruby
strip_entry_components!(entry)
next if entry_path_is_empty?(entry)
```

Reply:
yeah `!` at the end doesn't have fixed semantics like that, it just means it is a mutator or it throws or something.  returning nil while still mutating doesn't violate any idiom afaik.